common-lib/src/main/scala/com/gu/mediaservice/lib/aws/RoundedExpiration.scala

```diff
@@ -0,0 +1,13 @@
+package com.gu.mediaservice.lib.aws
+
+import org.joda.time.{DateTime, Duration}
+
+trait RoundedExpiration {
+
+  // Round expiration time to try and hit the cache as much as possible
+  // TODO: do we really need these expiration tokens? they kill our ability to cache...
+  def cachableExpiration: DateTime = roundDateTime(DateTime.now, Duration.standardMinutes(10)).plusMinutes(20)
+
+  private def roundDateTime(t: DateTime, d: Duration): DateTime = t minus (t.getMillis - (t.getMillis.toDouble / d.getMillis).round * d.getMillis)
```

Comment:
tangential at best to this PR so not blocking, but I stumbled over this for a while and I'm still not sure I fully get how it works, I came up with this alternative which does I think the same but a bit clearer

```
    val now = DateTime.now()
    val hour = now.getHourOfDay
    val minute = (now.getMinuteOfHour % 10) * 10
    val seconds = 0
    val millis = 0
    now.withTime(hour, minute, seconds, millis)
```

not flexible like the previous version, but then nothing else calls this so I'm relaxed about hardcoding 10 mins

Reply:
Yeah agree.
I'm tip toeing carefully around this without think about it too much because it's effective in a non CloudFront Grid install.
I'd need to visualise it properly but I expect it's grouping what would have been Date.now into the same block of length X beginning Y seconds from now.